MatchServer/MMatchServer.cpp

```diff
@@ -562,6 +568,7 @@ void MMatchServer::Destroy()
 	m_AsyncProxy.Destroy();
 	MGetMatchShop()->Destroy();
 	m_SafeUDP.Destroy();
+	MGetGunGame()->~GunGame();
```

Comment:
Apart from the fact that it shouldn't be there since you should never need to manually `delete` anything in modern C++, yeah, it looks correct.